process/docker.go

```diff
@@ -62,60 +48,41 @@ func (c *Command) String() string {
 }
 
 func (c *Command) Execute() (r *Result) {
-
-	var outPull io.ReadCloser
 	r = &Result{}
-	c.mutex.Lock()
-	defer c.mutex.Unlock()
-	outPull, r.SystemError = cli.ImagePull(c.ctx, "ubuntu", types.ImagePullOptions{})
-	if r.SystemError != nil {
-		return
-	}
-	defer outPull.Close()
-	io.Copy(c.writer, outPull)
-
-	c.resp, r.SystemError = c.cli.ContainerCreate(
-		c.ctx,
-		&container.Config{
-			Image:      "ubuntu",
-			Cmd:        c.cmd,
-			WorkingDir: c.workingDirectory,
-			Env:        c.env,
-		},
-		nil,
-		nil,
-		"",
-	)
-	if r.SystemError != nil {
-		return
-	}
 
-	r.SystemError = c.cli.ContainerStart(c.ctx, c.resp.ID, types.ContainerStartOptions{})
-	if r.SystemError != nil {
+	image := "ubuntu"
+
+	// TODO scary injection potential here
+	cmd := exec.CommandContext(c.ctx, "/usr/bin/docker", append([]string{"run", image}, c.cmd...)...)
```

Comment:
We should probably make the docker executable a configuration setting (e.g. on my mac it is `/usr/local/bin/docker`).

Reply:
Don't we need to pull the docker image before using it?

Reply:
Let's also log any commands we run to the worker log.

Reply:
We're logging commands now. Images get pulled as part of the `run` command, and are kept on disk for future use (not that we support that).